meshio/helpers.py

```diff
@@ -76,7 +76,23 @@
     '.vtk': 'vtk-binary',
     '.xdmf': 'xdmf',
     '.xmf': 'xdmf',
-    }
+}
+
+
+def check_filename(filename, file_format=None):
+    # Checks the input and output file name and format.
+    # https://stackoverflow.com/q/4843173/353337
+    assert isinstance(filename, str), 'Invalid input \'{}\''.format(filename)
+
+    if not file_format:
+        # deduce file format from extension
+        extension = '.' + os.path.basename(filename).split(os.extsep, 1)[-1]
+        # check whether the file format is supported
+        assert extension in _extension_to_filetype, ('Unknown file extension \'{}\' of \'{}\'.'
+                                                     .format(extension, filename))
+        file_format = _extension_to_filetype[extension]
+
+    return filename, file_format
```

Comment:
Does this really need to return the `filename`? You could call this function `format_from_filename`). Also, the `file_format` argument is somewhat weird.

Reply:
If the user specifies a different extension in `filename` and `file_format` then currently the code gives preference to `file_format` but while writing the output gives extension given in the `filename`. This could sometime lead to confusion.

I made the above function to check the following:

- Whether the input filename is a string.
- If the file_format is not specified then get the file_format from the filename.
- If there is a conflict between the formats specified via extension and file_format, then, give preference to the user-specified file_format.

The final check was not made since the `read` and `write` method handle `file_format` in different ways. Thus, in its current state, the function doesn't require to return `filename` but has to return `file_format`.

Reply:
Fair enough, but you still don't need to return `filename`, right? It's never changed.

Reply:
Also, the `None` check should happen outside the function for clarity, i.e.,
```python
if file_format is None:
    file_format = format_from_filename(filename)
```
This is much clearer.

Reply:
But if we do this way then we have to separately make the filename is a string check in both read and write. In the current form, we can leave the check inside the format_from_filename and then the functionality would be as follows:
- If the user has specified a format then the function would check for the validity of the filename only.
- If he has not specified a format then the function would get the format from the extension.

Reply:
> But if we do this way then we have to separately make the filename is a string check in both read and write

I don't get it. Why?

Reply:
Because if the user specifies a `file_format` then the code would not check whether the filename is a string as it would not call the function. So, we need to check separately in both `read` and `write`.